merlin-sdk/src/main/java/gov/nasa/jpl/aerie/merlin/protocol/driver/Scheduler.java

```diff
@@ -1,15 +1,13 @@
 package gov.nasa.jpl.aerie.merlin.protocol.driver;
 
 import gov.nasa.jpl.aerie.merlin.protocol.model.Task;
-import gov.nasa.jpl.aerie.merlin.protocol.types.SerializedValue;
-
-import java.util.Map;
 
 public interface Scheduler {
```

Comment:
General thought - I wonder if `Scheduler` is the best word for the interface between the model and the simulation engine. It is certainly responsible for requesting that child tasks be scheduled, but `get` and `emit` don't seem to have to do with scheduling, at least not from the caller's perspective. (Full disclosure, I am motivated to think about this because I spend time thinking about the other thing called a "Scheduler" in the aerie system)

Reply:
I originally named this in reference to [process schedulers in operating systems](https://en.wikipedia.org/wiki/Scheduling_(computing)#Process_scheduler), but I totally see the collision. It's more of a set of exposed syscalls than the scheduler itself.

I'm certainly open to hearing alternative names 🤔

Reply:
I do like the operating system analogy... No great names come to mind, I'll keep this in the back of my mind though. Interestingly enough none of the four methods on `Scheduler` actually cause the task to yield - that's done by returning a status from `step`, so the cooperative multitasking is done by a different mechanism 🤔 On my current train of thought, the Scheduler seems to be less about multitasking and more about querying and perturbing the state of the simulation...

This is a discussion to be had separately from this PR, I just wanted to note down the ideas that passed through my head as I read through these commits 😃

Reply:
I definitely agree; the analogy isn't perfect, if it ever was. I think it's because `spawn` is part of this interface? 🤔

But yes, it's a good discussion to have 😄